packages/loader/container-loader/src/container.ts

```diff
@@ -1137,12 +1137,11 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         if (this.service === undefined) {
             throw new Error("Not attached");
         }
-        const storageService = await this.service.connectToStorage();
+        let service = await this.service.connectToStorage();
 
         // Enable prefetching for the service unless it has a caching policy set otherwise:
-        const service = new PrefetchDocumentStorageService(storageService);
-        if (this.service.policies?.caching === LoaderCachingPolicy.NoCaching) {
-            service.stopPrefetch();
+        if (this.service.policies?.caching !== LoaderCachingPolicy.NoCaching) {
+            service = new PrefetchDocumentStorageService(service);
```

Comment:
why not just return this here, and not need a mutable var

Reply:
Jatin adds one more case like that in another PR, so I'd prefer all of them to be an adapter wrappers until we are done and just return result. Such that we can change order of wrappers (in case it starts to matter for some reason) easily.
No big deal either way - I can change it if you feel it's subjectively better.

Reply:
If we're amending the grammar, I would prefer a full rewording so that it reads overall like we assume no caching by default, i.e. something like:
```
const service = await this.service.connectToStorage();
if (policies.caching === LoaderCachingPolicy.Caching) {
return new Prefetch(service);
}
return service;
```
But this would be bigger change.

Reply:
I'd prefer it to be independent block of code that can be anywhere in this function, across similar wrappers/adapters.